controllers/clustergroupupgrade_controller.go

```diff
@@ -376,9 +376,22 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
 									})
 								} else {
 									r.Log.Info("Batch upgrade timed out")
-									clusterGroupUpgrade.Status.Status.CurrentBatchStartedAt = metav1.Time{}
-									if clusterGroupUpgrade.Status.Status.CurrentBatch < len(clusterGroupUpgrade.Status.RemediationPlan) {
-										clusterGroupUpgrade.Status.Status.CurrentBatch++
+									switch clusterGroupUpgrade.Spec.BatchTimeoutAction {
+									case ranv1alpha1.BatchTimeoutAction.Abort:
+										// If the value was abort then we need to fail out
+										meta.SetStatusCondition(&clusterGroupUpgrade.Status.Conditions, metav1.Condition{
+											Type:    "Ready",
+											Status:  metav1.ConditionFalse,
+											Reason:  "UpgradeTimedOut",
+											Message: "The ClusterGroupUpgrade CR policies are taking too long to complete",
+										})
+									case ranv1alpha1.BatchTimeoutAction.Continue:
```

Comment:
What you are going to do in this case ranv1alpha1.BatchTimeoutAction.Continue: ? if nothing is set or don, shouldn't the default case suffice?

Reply:
The default case would suffice but I just like being a bit explicit. I can just use the default case if you want though

Reply:
I would not keep a case if we are not performing any action on that case.

Reply:
already updated it in the latest commit